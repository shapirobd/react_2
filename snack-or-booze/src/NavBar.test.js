import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import NavBar from "./NavBar";

test("it should render without crashing", () => {
	render(
		<MemoryRouter>
			<NavBar />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<NavBar />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

// test("should contain a link to add new item", () => {
// 	const { getByText } = render(
// 		<MemoryRouter>
// 			<NavBar />
// 		</MemoryRouter>
// 	);
// 	expect(getByText("Add")).toBeInTheDocument();
// });

// ^^^ it keeps saying .toBeInTheDocument() is not a function when i try to run tests - spent a lot of time trying to fix it but I just can't figure it out
