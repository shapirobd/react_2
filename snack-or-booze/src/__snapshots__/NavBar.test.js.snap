// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Snack or Booze
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/add"
          >
            Add
          </a>
          <a
            href="/snacks"
          >
            Snacks
          </a>
          <a
            href="/drinks"
          >
            Drinks
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
